### PR TITLE
[WIP][#1900] Refactor inline event handlers

### PR DESCRIPTION
**Pull request by mauriel133**
I have refactored all occurrences of inline event handlers in our scripts. Next, we should test all the instances that were refactored to make sure nothing broke.

I tested many of them and made sure they work, but for others I'm not sure of the workflow. I have left `TODO` comments on the ones that still need to be tested.

Developers that worked on resource types should verify that the refactored code works as expected.

This code is deployed at https://park.hydroshare.org
User: park
Password: park

---------------------
Known issues:

- [ ] Failing Django-generated script
image
- [ ] Unable to load external images for app icons.
image

### CODE REVIEW

**Comment by aphelionz** [comment]
🎉🎉🎉🖒🖒🖒

Will review soon

**Comment by mjstealey** [comment]
Flake8 Fun
```
./hs_app_netCDF/forms.py:220:101: E501 line too long (107 > 100 characters)
./hs_app_timeseries/forms.py:743:101: E501 line too long (117 > 100 characters)
./hs_core/forms.py:587:101: E501 line too long (112 > 100 characters)
./hs_file_types/models/base.py:400:101: E501 line too long (119 > 100 characters)
./hs_file_types/models/base.py:451:101: E501 line too long (106 > 100 characters)
./hs_file_types/models/generic.py:47:87: E262 inline comment should start with '# '
./hs_file_types/models/generic.py:58:87: E262 inline comment should start with '# '
./hs_file_types/models/raster.py:79:101: E501 line too long (105 > 100 characters)
./hs_geo_raster_resource/forms.py:300:101: E501 line too long (105 > 100 characters)
./hs_modflow_modelinstance/forms.py:531:101: E501 line too long (126 > 100 characters)
```

**Comment by mauriel133** [comment]
@gantian127 , @pkdash 

This code is now ready for testing and review at https://testservices.hydroshare.org/
User: park
Password: park

Please list any issues you find in this thread.

**Comment by aphelionz** [comment]
Do fix the merge conflict, ofc.

**Comment by pkdash** [comment]
@aphelionz 

@mauriel133 wanted me to do some manual testing of these changes. I have not done any testing yet. I think it would be good to do some testing prior to merging as we don't have any automated tests for these kind of changes. I am not sure if @gantian127 has done any testing.

**Comment by mauriel133** [comment]
We need to verify that all handlers work. I tested most of them. The ones I could not test have a comment on them: "TODO: TESTING"

File: theme/static/js/my-resources.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        "searchable": true
+    },
+    {
+        "targets": [SHARING_STATUS_COL],     // Sharing status
+        "visible": false,
+        "searchable": false
+    }
+];
+// TODO: fix the following references
+// $(document).ready(function () {
```
Comment (pkdash) [inline theme/static/js/my-resources.js]: Should this commented code be removed/

Comment (mauriel133) [inline theme/static/js/my-resources.js]: This code was commented out in another issue regarding unit tests. I believe there's pending work to be done for this in another ticket.

**Comment by pkdash** [comment]
@mauriel133 These conflicts need to be resolved.

**Comment by pkdash** [comment]
@mauriel133 Seems there are conflicts that need to be resolved. Hope this PR gets merged soon.

**Comment by mauriel133** [comment]
@pkdash, this will keep having merge conflicts until the testing is done and we merge. We need to finish testing the items listed in #1935. After that is done I will fix the merge conflicts and merge this PR.

**Comment by pkdash** [comment]
@dtarb Was wondering if we still need to wait for the remaining tests to be done. I suggest @mauriel133 should merge this after fixing the conflicts without further wait.

**Comment by dtarb** [comment]
I just sent a reminder to @vmmerwade and @xuezhaokun about these tests, but park.hydroshare.org where they are supposed to test is not up.

**Comment by dtarb** [comment]
@pkdash the manual checks have been done.

**Comment by pkdash** [comment]
@mauriel133 It seems the manual testing is complete at this point. Can you resolve the conflicts and merge after Jenkins checks are successful? Thanks.

**Comment by mauriel133** [comment]
I have fixed the merge conflicts. I will be merging this after Jenkins tests pass.

**Comment by mauriel133** [comment]
@aphelionz, @pkdash it looks like we have a failing test.

> TypeError: create_bag_files() got an unexpected keyword argument 'fed_zone_home_path'

Any ideas on how to fix this?

**Comment by pkdash** [comment]
@mauriel133 It seems your branch is using an older version of django_irods. Do you know how to update that?

**Comment by mauriel133** [comment]
I have updated django_irods now. Merging.

**Pull request merged**